packages/tupaia-web-server/src/__tests__/errors.test.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { TestableServer } from '@tupaia/server-boilerplate';
+import { setupTestApp } from './testUtilities';
+
+describe('Error responses', () => {
+  let app: TestableServer;
+
+  beforeAll(async () => {
+    app = await setupTestApp();
+  });
+
+  describe('Microservice errors', () => {
+    it('Requesting an entity with no permissions', async () => {
```

Comment:
I like the idea behind this, but with these orchestration servers we don't really need to test routes that are forwarded, otherwise we are doubling up on testing, once in the orchestration server and once in the actual server.

If we want to add coverage here we should test that the api client mock has had a specific method called on the respective client, here it would be apiClientMock.entity.getEntity. Up to you if we want to do this, I'm happy enough not covering forwarded routes, we still have a lot of core business logic that doesn't have tests yet that might warrant the attention more.

Reply:
Yes and no... Perhaps I need to rephrase this test case

Prior to recent changes, the orchestration server would consume any error from a microservice and return `500` regardless of what the original error was. This unit test exists to confirm that the server is now forwarding errors correctly. It's testing the behaviour of the orchestration server, not the microservice

Reply:
I see. Yeah good thinking and thanks for adding that test!

I think keeping the MockTupaiaApiClient simple and happy path is a good idea. 

I think a simple jest mock like this would be a good way to test non-happy path stuff: https://github.com/beyondessential/tupaia/commit/3037becf5555106e3553b5fd028cd14e610c704d

From the docs there does seem to be a way of defining a global module mock and then overriding the implementation on a per-test basis, but I couldn't figure it out. The other way we've done it in the past to avoid global mocks is with a test helper like setupApiClientMock() which I left unimplemented in the example.

Reply:
Cheers @IgorNadj , I've implemented some of those suggestions (although I find splitting out the `handleServerError` function a bit unnecessary)

Also reverted the changes to the mock entity api